### PR TITLE
add missing unit tests for #2108

### LINKED ISSUE #2226: Add missing unit test for token middleware
as discussed in #2108

### PR DESCRIPTION
**Pull request by benkroeger**
subsequent token middleware tries to read `token.id` when `enableDoublecheck: true`. That caused a "Cannot read property `id` of `null`" error when the first middleware didn't actually find a valid accessToken.

@bajtos 

fixes #2226

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
@slnode ok to test

File: test/access-token.test.js [hunk 1]
```diff
@@ -249,6 +249,39 @@ describe('loopback.token(options)', function() {
...
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
+      });
+
+      request(app).get('/')
+        .set('Authorization', token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body).to.eql({
```
Comment (bajtos) [inline test/access-token.test.js]: Is it necessary to check all token properties here? I am worried that if we add a new Token property in the future, we will have to fix way too many tests that don't really depend on specific Token properties.

It think it is enough to check that `res.body.id` is the same as `token.id`, isn't it?

File: test/access-token.test.js [hunk 2]
```diff
@@ -249,6 +249,39 @@ describe('loopback.token(options)', function() {
...
+            id: token.id,
+            ttl: token.ttl,
+            userId: token.userId,
+            created: token.created.toJSON(),
+          });
+          done();
+        });
+    });
```
Comment (bajtos) [inline test/access-token.test.js]: Please add an empty line after the test case to serve as a visual delimiter.

**Comment by bajtos** [comment]
@benkroeger thank you for the pull request. The patch looks good in general, I pointed out two details to improve in the comments above.

File: test/access-token.test.js (hunk 1 above)
Comment (benkroeger) [inline test/access-token.test.js]: in my last unit test, I did exactly that - and you commented that only checking on `id` wouldn't be enough.
Let me know which one you prefer

Comment (bajtos) [inline test/access-token.test.js]: I see, sorry for me not being consistent! Let's keep what you have now.

**Comment by bajtos** [comment]
Landed via 103935c, thank you for the contribution!

**Pull request closed without merging**